rest/config.go

```diff
@@ -1046,6 +1042,26 @@ func RunServer(config *ServerConfig) {
 	config.Serve(*config.Interface, CreatePublicHandler(sc))
 }
 
+func validateAndAddDBFromConfig(sc *ServerContext, databases DbConfigMap) (warnings []string) {
```

Comment:
This function is doing too much to be easily tested - a similiar problem to when all of this was inside RunServer.

I'd advise breaking it down into the following:

- Have a function that adds a bunch of databases from a `DbConfigMap`, which returns:
  1. The full map of bucket UUIDs to DatabaseContexts.
  2. The error for the first one that can't be opened.

- Have a second function that takes in the bucket UUID to DatabaseContext map, does the duplicate checks (you can do a simple length check of the dbContext slice) and returns which ones were duplicates.

- RunServer can now iterate over the duplicates, and do the logging. While your test can assert on the duplicates returned.

Reply:
This has been implemented and changes are committed.